### PR TITLE
refactor: Remove unused authZ provider logic

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Remove unused authZ provider logic
* Deployed and tested login, add user logic were not affected

Checklist: 

* [x] Have you successfully deployed to an AWS account with your changes? 
* [x] Have you linted your code locally prior to submission?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js [hunk 1]
```diff
@@ -74,22 +68,6 @@ const AuthenticationProviderConfigsStore = BaseStore.named('AuthenticationProvid
       });
     },
 
-    getUpdateAuthenticationProviderConfigEditor(authenticationProviderConfigId) {
```
Comment (SanketD92) [inline addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js:77]: This method is being called by `addons/addon-base-ui/packages/base-ui/src/parts/authentication-providers/EditAuthenticationProvider.js` (line 95). Do we not need changes there?

File: addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js [hunk 2]
```diff
@@ -114,15 +92,6 @@ const AuthenticationProviderConfigsStore = BaseStore.named('AuthenticationProvid
       return undefined;
     },
 
-    async updateAuthenticationProvider(authenticationProviderConfig) {
```
Comment (SanketD92) [inline addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js:117]: Same comment as above.

File: addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js (hunk 1 above)
Comment (Bingjiling) [reply addons/addon-base-ui/packages/base-ui/src/models/authentication/AuthenticationProviderConfigsStore.js:77]: Good catch! Will update.

**Comment by Bingjiling** [comment]
Hi @SanketD92, thanks for the comments earlier. I did another review of the front end functionality and realized the auth provider update method is actually in use by the front end, and it does update the DDB for the auth provider information. However, the update will only stay temporarily till the next deployment is done. I've added the codes related to update auth provider back, and only removed the endpoint for create auth provider, which was not used by any front end assets. Meanwhile, I'll create another ticket to refactor the update method since we need to understand how/if customers are using this method before we do anything.

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**